dipy/align/imaffine.py

```diff
@@ -566,6 +575,29 @@ def setup(self, transform, static, moving, static_grid2world=None,
         self.affine_map = AffineMap(P, static.shape, static_grid2world,
                                     moving.shape, moving_grid2world)
 
+        # Masks can only be used with dense sampling
+        if (self.sampling_proportion is None or
+                self.sampling_proportion == 1.0):
+
+            if static_mask is not None:
+                self.static_mask = static_mask.astype(np.int32)
+            else:
+                self.static_mask = None
+
+            if moving_mask is not None:
+                self.moving_mask = moving_mask.astype(np.int32)
+            else:
+                self.moving_mask = None
```

Comment:
same comment as above: Not sure, this `else` is necessary

Reply:
I disagree, this would leave self.moving_mask undefined, but this value must be defined.